src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -326,55 +327,56 @@ static bool GetDescriptionValue(
         }
 
         *value = valueStart;
-        valueLength = index;
+        *valueLength = index;
         return true;
     }
 
     return false;
 }
 
+#define descriptionLength 256
+
 /*
 Parses the Kx, Enc, and Mac values out of the SSL_CIPHER_description and
 maps the values to the corresponding .NET enum value.
 */
 static bool GetSslConnectionInfoFromDescription(const SSL_CIPHER* cipher,
-                                                CipherAlgorithmType& dataCipherAlg,
-                                                ExchangeAlgorithmType& keyExchangeAlg,
-                                                HashAlgorithmType& dataHashAlg,
-                                                DataHashSize& hashKeySize)
+                                                CipherAlgorithmType* dataCipherAlg,
+                                                ExchangeAlgorithmType* keyExchangeAlg,
+                                                HashAlgorithmType* dataHashAlg,
+                                                DataHashSize* hashKeySize)
 {
-    const int descriptionLength = 256;
-    char description[descriptionLength] = {};
+    char description[descriptionLength];
     SSL_CIPHER_description(cipher, description, descriptionLength - 1); // ensure description is NULL-terminated
```

Comment:
Previously the whole array was being initialized to zero, which is how it was ensuring that it was NULL-terminated.  After your change, that's no longer the case.

Reply:
Great, I have fixed it by adding `= {0}`.

In C++, empty brace `{}` is allowed but not in C as noted in http://en.cppreference.com/w/c/language/array_initialization. In C and C++ both, `= {0}` is a valid syntax. However, in C's global scope (outside of function body), declaration without initialization zeros out the array: https://stackoverflow.com/a/2589751.